exporter/datadogexporter/go.mod

```diff
@@ -4,15 +4,15 @@ go 1.20
 
 require (
 	github.com/DataDog/agent-payload/v5 v5.0.102
-	github.com/DataDog/datadog-agent/pkg/proto v0.50.2
+	github.com/DataDog/datadog-agent/pkg/proto v0.52.0-devel
 	github.com/DataDog/datadog-agent/pkg/trace v0.50.2
```

Comment:
nit: do you want to upgrade this one to `v0.52.0-devel` too?

Reply:
I didn't explicitly bump the other one, it was just because of https://github.com/DataDog/opentelemetry-mapping-go/blob/81d9650bebbbcff44cec39ac97fd9c46420c32bc/pkg/otlp/metrics/go.mod#L6

But it makes sense to bump the rest as well, let me do that

Reply:
Done in 6b4be71

Reply:
Looks like this surfaced some problem: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7585607691/job/20662004468?pr=30680#step:9:166

I'm not sure whether we want to resolve that in this PR or revert the dep upgrade, WDYT?

Reply:
I'll revert the bump and open a separate PR for this